ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/AEMFutureRunnable.java

```diff
@@ -0,0 +1,17 @@
+package org.cru.godtools.articles.aem.service;
+
+import android.content.Context;
+
+public class AEMFutureRunnable implements Runnable {
+    private AEMDownloadManger mManager;
+
+    public AEMFutureRunnable(Context context) {
+        mManager = AEMDownloadManger.getInstance(context);
+    }
+
+    @Override
+    public void run() {
+        mManager.extractAemImportsFromManifestsTask();
+        mManager.syncStaleAemImportsTask();
```

Comment:
@frett I created this class so I didn't have to change the access modifier for any methods.  Let me know if this and the way I handled the Future Task are the headed in the right direction.

Reply:
I was actually thinking of relaxing the visibility of the necessary enqueue methods and have them return a `ListenableFuture`.

Reply:
also, the swipe to refresh should be a targeted force sync, not a sync everything

Reply:
so, it should end up more directly calling `enqueueSyncAemImport(uri, true)`

Reply:
the task scheduling methods are meant for external consumption, the actual task methods are meant for internal use only. The reason I made them package-private and not private is to prevent creation of synthetic accessor methods.

See: https://medium.com/thoughts-overflow/effects-of-javas-synthetic-accessor-methods-in-android-bb67b3bac22e

Reply:
it's a minor thing that helps keep method counts low while we still have to deal with old android and want to avoid multidex if possible.

Reply:
@frett not sure what you mean by a targeted forced sync can you clarify?

Reply:
since you are viewing a list of articles for a specific manifest, you should only sync the AEM Imports that the manifest contains